packages/devp2p/src/types.ts

```diff
@@ -237,3 +242,57 @@ export interface RLPxOptions {
 }
 
 export type SendMethod = (code: number, data: Uint8Array) => any
+
+export interface RLPXEvents {
+  'peer:added': Peer
+  'peer:error': { peer: Peer; error: any }
+  'peer:removed': { peer: Peer; reason: any; disconnectWe: any }
+  error: Error
+  close: undefined
+  listening: undefined
+}
+
+export interface PeerEvents {
+  error: Error
+  connect: undefined
+  close: { reason: any; disconnectWe: any }
+}
+
+export interface ProtocolEvents {
+  message: {
+    code: SNAP.MESSAGE_CODES | ETH.MESSAGE_CODES | LES.MESSAGE_CODES
+    payload: Uint8Array | NestedUint8Array
+  }
+  status:
+    | LES.Status
+    | {
+        chainId: Uint8Array | Uint8Array[]
+        td: Uint8Array
+        bestHash: Uint8Array
+        genesisHash: Uint8Array
+        forkId?: Uint8Array | Uint8Array[]
+      }
+}
```

Comment:
We normally type/name everything in singular, can we do this here too?

Otherwise this all looks super great! 🎉👍🙂

Reply:
What do you mean by type in singular?  For our current event typing, it's all `VMEvents` or `EVMEvents`.  I have no strong opinion but was trying to be consistent with our existing event typing.

Reply:
Ah, ok, then I guess `VMEvents` also made no sense in the first place.

So, such a thing is clearly *one* event:

```ts
export interface EVMEvents {
  newContract: { data: NewContractEvent; resolve?: (result?: any) => void }
  beforeMessage: { data: Message; resolve?: (result?: any) => void }
  afterMessage: { data: EVMResult; resolve?: (result?: any) => void }
  step: { data: InterpreterStep; resolve?: (result?: any) => void }
}
```

And we also just type anything else in the monorepo with an anlogue structure with "singular", so these events clearly stand out.

Anyhow: don't want to bring you on a side track then, will open a small refactoring issue on this.

Reply:
(or maybe you can directly adjust here for `VMEvents` and `EVMEvents`, at the end it's a pretty small update?)

Reply:
Yep, it's a simple change and makes sense to do now.